x-pack/plugin/vectors/src/test/java/org/elasticsearch/xpack/vectors/query/DenseVectorFunctionTests.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.vectors.query;
+
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.Version;
+import org.elasticsearch.script.ScoreScript;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.vectors.query.ScoreScriptUtils.CosineSimilarity;
+import org.elasticsearch.xpack.vectors.query.ScoreScriptUtils.DotProduct;
+import org.elasticsearch.xpack.vectors.query.ScoreScriptUtils.L1Norm;
+import org.elasticsearch.xpack.vectors.query.ScoreScriptUtils.L2Norm;
+import org.elasticsearch.xpack.vectors.query.VectorScriptDocValues.DenseVectorScriptDocValues;
+import org.junit.Before;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.elasticsearch.xpack.vectors.mapper.VectorEncoderDecoderTests.mockEncodeDenseVector;
+import static org.hamcrest.Matchers.containsString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class DenseVectorFunctionTests extends ESTestCase {
+    private String field;
+    private float[] docVector;
+    private List<Number> queryVector;
+    private List<Number> invalidQueryVector;
+
+    @Before
+    public void setUpVectors() {
+        field = "vector";
+        docVector = new float[] {230.0f, 300.33f, -34.8988f, 15.555f, -200.0f};
+        queryVector = Arrays.asList(0.5f, 111.3f, -13.0f, 14.8f, -156.0f);
+        invalidQueryVector = Arrays.asList(0.5, 111.3);
+    }
+
+    public void testDenseVectorFunctions() {
+        for (Version indexVersion : Arrays.asList(Version.V_7_4_0, Version.CURRENT)) {
```

Comment:
We have a loop here as we expect a change in every version?

Reply:
I wasn't planning to add more versions (unless we have more version-dependent changes). The loop is really just to test these two listed versions.